docs/src/main/asciidoc/mongodb-panache.adoc

```diff
@@ -387,6 +387,53 @@ public class Person extends PanacheMongoEntity {
 Both `findByNameWithPanacheQLQuery()` and `findByNameWithNativeQuery()` methods will return the same result but query written in PanacheQL
 will use the entity field name: `name`, and native query will use the MongoDB field name: `lastname`.
 
+== Query projection
+
+Query projection can be done with the `project(Class)` method on the `PanacheQuery` object that is returned by the `find()` methods.
+
+You can use it to restrict which fields will be returned by the database,
+the ID field will always be returned but it's not mandatory to include it inside the projection class.
+
+For this, you need to create a class (a Pojo) that will only contains the projected fields.
+This pojo needs to be annotated with `@ProjectionFor(Entity.class)` where `Entity` is the name of your entity class.
+The field names, or the getters, of the projection class will be used to restrict which properties will be loaded from the database.
+
+Projection can be done for both PanacheQL and native queries.
+
+[source,java]
+----
+import io.quarkus.mongodb.panache.ProjectionFor;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+// using public fields
+@ProjectionFor(Person.class)
+public class PersonName {
+    public String name;
+}
+
+// using getters
+@ProjectionFor(Person.class)
+public class PersonNameWithGetter {
+    private String name;
+
+    public String getName(){
+        return name;
+    }
+
+    public void setName(String name){
+        this.name = name;
+    }
+}
+
+// only 'name' will be loaded from the database
+PanacheQuery<PersonName> shortQuery = Person.find("status ", Status.Alive).project(PersonName.class);
+PanacheQuery<PersonName> query = Person.find("'status': ?1", Status.Alive).project(PersonNameWithGetter.class);
+PanacheQuery<PersonName> nativeQuery = Person.find("{'status': 'ALIVE'}", Status.Alive).project(PersonName.class);
+----
+
+TIP: You can see that there is no need to use `@BsonProperty` annotations to define custom column mappings.
```

Comment:
I have re-read it a couple of times this tip and I find it hard to follow. Is there some ay we could re-phrase it? 

Otherwise, really great stuff. Thanks.

Reply:
> Using `@BsonProperty` is not needed to define custom column mappings, as the mappings from the entity class will be used.

This is clearer and more detailed

Reply:
Go for it.

Reply:
updated